types/sinon/index.d.ts

```diff
@@ -234,8 +234,8 @@ declare namespace Sinon {
 
     interface SinonFakeTimersStatic {
         (): SinonFakeTimers;
-        (...timers: string[]): SinonFakeTimers;
-        (now: number, ...timers: string[]): SinonFakeTimers;
+        (now?: number | Date): SinonFakeTimers;
```

Comment:
i suppose it doesn't need to be optional because of the previous declaration?

Reply:
I agree this should probably be adjusted. Based on advice from [Typescript Do's and Don'ts](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html) and on the sinon docs I think we can actually have just a single declaration with `(arg?: { ... config fields ... } | number)`

I don't think this is a big deal since it's just an interface but Typescript's overload resolution rules mean that in general having fewer overloads due to using optional parameters can pay off. This is closer in spirit to how the argument evaluation in Javascript will work anyway but that may not be a plus from a Typescript perspective.

Reply:
@sommestad what do you think about this? Admittedly it may be a bit of a nitpick and you could argue that the current set of 3 separate overloads expresses the sinon spec more clearly.

Reply:
@sommestad I agree that this should be changed to use optional parameters as @rationull suggested.